### PR TITLE
Adds List Workflows for Python

### PR DESCRIPTION
**Pull request by rachfop**
## What does this PR do?

Adds "List Workflows" to the Python dev guide under observability.

### CODE REVIEW

File: docs-src/python/how-to-list-workflow-executions-using-the-client-in-python.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+id: how-to-list-workflow-executions-using-the-client-in-python
```
Comment (cretz) [inline docs-src/python/how-to-list-workflow-executions-using-the-client-in-python.md:2]: I don't see https://github.com/temporalio/documentation/blob/96672716e70f49372fafa66bf1d58ba7ba0fea1a/docs/application-development/observability.md?plain=1#L686-L692 updated in this PR. Doesn't this need to be referenced in the guide?

File: docs-src/python/how-to-list-workflow-executions-using-the-client-in-python.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  - sdk
+  - python
+---
+
+Use the [list_workflows()](https://python.temporal.io/temporalio.client.Client.html#list_workflows) method and pass a [List Filter](/concepts/what-is-a-list-filter) as an argument to filter the listed Workflows.
+
+```python
+client = await Client.connect("localhost:7233")
+client.list_workflows('TaskQueue="your-activity-task-queue"')
+client.list_workflows('WorkflowId="your-activity-workflow-id"')
```
Comment (cretz) [inline docs-src/python/how-to-list-workflow-executions-using-the-client-in-python.md:17]: Just calling this is not a good enough example. You need to show something like:

```python
async for workflow in client.list_workflows('WorkflowType="MyWorkflowClass"'):
    print(f"Workflow: {workflow.id}")
```

File: docs-src/python/how-to-list-workflow-executions-using-the-client-in-python.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  - developer-guide
+  - sdk
+  - python
+---
+
+Use the [list_workflows()](https://python.temporal.io/temporalio.client.Client.html#list_workflows) method and pass a [List Filter](/concepts/what-is-a-list-filter) as an argument to filter the listed Workflows.
+
+```python
+client = await Client.connect("localhost:7233")
+client.list_workflows('TaskQueue="your-activity-task-queue"')
```
Comment (cretz) [inline docs-src/python/how-to-list-workflow-executions-using-the-client-in-python.md:16]: `TaskQueue` is not a standard visibility field, so I wouldn't recommend using it in the docs

File: docs-src/python/how-to-list-workflow-executions-using-the-client-in-python.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+title: How to list Workflows in Python
+sidebar_label: List Workflows
+description: List Workflows from the Client using `list_workflows()`.
+tags:
+  - developer-guide
+  - sdk
+  - python
+---
+
+Use the [list_workflows()](https://python.temporal.io/temporalio.client.Client.html#list_workflows) method and pass a [List Filter](/concepts/what-is-a-list-filter) as an argument to filter the listed Workflows.
```
Comment (cretz) [inline docs-src/python/how-to-list-workflow-executions-using-the-client-in-python.md:12]: Can you confirm the rendered link of `/concepts/what-is-a-list-filter` becomes `/visibility#list-filter`?

File: docs-src/python/how-to-list-workflow-executions-using-the-client-in-python.md (hunk 2 above)
Comment (cretz) [inline docs-src/python/how-to-list-workflow-executions-using-the-client-in-python.md:17]:
```suggestion
client.list_workflows('WorkflowId="your-workflow-id"')
```

**Review by cretz — APPROVED** [review]

**Review by aarohib — APPROVED** [review]

**Pull request merged**